core/src/test/scala/org/bykn/bosatsu/LetFreeTest.scala

```diff
@@ -56,14 +68,87 @@ package Recur/Some
 
 out = foo
-"""
-      ), "Recur/Some", Recursion(Lambda(Lambda(Match(LambdaVar(0),NonEmptyList.of(
-        (PositionalStruct(Some(0), Nil),Struct(1,List(Literal(Str("a")), Struct(1,List(Literal(Str("b")), Struct(1,List(Literal(Str("c")), Struct(0,List())))))))),
-      (PositionalStruct(Some(1), List(WildCard, Var(0))),Lambda(Struct(1,List(Literal(Str("zero")), App(LambdaVar(2),LambdaVar(0))))))
-      )))))
+"""),
+      "Recur/Some",
+      Recursion(
+        Lambda(
+          Lambda(
+            Match(
+              LambdaVar(0),
+              NonEmptyList.of(
+                (
+                  PositionalStruct(Some(0), Nil, Enum),
+                  Struct(
+                    1,
+                    List(
+                      Literal(Str("a")),
+                      Struct(
+                        1,
+                        List(
+                          Literal(Str("b")),
+                          Struct(
+                            1,
+                            List(Literal(Str("c")), Struct(0, List(), Enum)),
+                            Enum
+                          )
+                        ),
+                        Enum
+                      )
+                    ),
+                    Enum
+                  )
+                ),
+                (
+                  PositionalStruct(
+                    Some(1),
+                    List(
+                      StrPat(
+                        NonEmptyList
+                          .of(StrPart.LitStr("a: "), StrPart.NamedStr(0))
+                      ),
+                      Var(1)
+                    ),
+                    Enum
+                  ),
+                  Lambda(
+                    Lambda(
+                      Struct(
+                        1,
+                        List(LambdaVar(0), App(LambdaVar(3), LambdaVar(1))),
+                        Enum
+                      )
+                    )
+                  )
+                ),
+                (
+                  PositionalStruct(Some(1), List(WildCard, Var(0)), Enum),
+                  Lambda(
+                    Struct(
+                      1,
+                      List(
+                        Literal(Str("zero")),
+                        App(LambdaVar(2), LambdaVar(0))
+                      ),
+                      Enum
+                    )
+                  )
+                )
+              )
+            )
+          )
+        )
+      ),
+      List({ lfe: LetFreeExpression =>
+        assert(
+          lfe.serialize ==
+            "Recursion(Lambda(Lambda(Match(LambdaVar(0),PositionalStruct(0,),Struct(1,Literal('a'),Struct(1,Literal('b'),Struct(1,Literal('c'),Struct(0,)))),PositionalStruct(1,StrPat(LitStr(a: ),NamedStr(0)),Var(1)),Lambda(Lambda(Struct(1,LambdaVar(0),App(LambdaVar(3),LambdaVar(1))))),PositionalStruct(1,WildCard,Var(0)),Lambda(Struct(1,Literal('zero'),App(LambdaVar(2),LambdaVar(0))))))))",
```

Comment:
as I mentioned above, can we just hash this and print the hex or something? Leave the string in a comment?